packages/indexer-agent/src/agent.ts

```diff
@@ -40,6 +43,16 @@ const deploymentInList = (
 ): boolean =>
   list.find(item => item.bytes32 === deployment.bytes32) !== undefined
 
+const deploymentRuleInList = (
```

Comment:
@hopeyen I've created this function to check if a deployment based rule already exists in the db. I use this below in `convertSubgraphBasedRulesToDeploymentBased` to check for existing deployment rules before creating them.  This way any user defined `deployment` rules take precedent over `subgraph` based rules. I'll make note of this order of precedent in the documentation.